crates/next-core/src/next_manifests/mod.rs

```diff
@@ -16,16 +25,112 @@ pub struct PagesManifest {
     pub pages: HashMap<RcStr, RcStr>,
 }
 
-#[derive(Serialize, Default, Debug)]
-#[serde(rename_all = "camelCase")]
+#[derive(Debug, Default)]
 pub struct BuildManifest {
-    pub dev_files: Vec<RcStr>,
-    pub amp_dev_files: Vec<RcStr>,
-    pub polyfill_files: Vec<RcStr>,
-    pub low_priority_files: Vec<RcStr>,
-    pub root_main_files: Vec<RcStr>,
-    pub pages: HashMap<RcStr, Vec<RcStr>>,
-    pub amp_first_pages: Vec<RcStr>,
+    pub polyfill_files: Vec<ResolvedVc<Box<dyn OutputAsset>>>,
+    pub root_main_files: Vec<ResolvedVc<Box<dyn OutputAsset>>>,
+    pub pages: FxIndexMap<RcStr, Vc<OutputAssets>>,
+}
+
+impl BuildManifest {
+    pub async fn build_output(
+        self,
+        output_path: Vc<FileSystemPath>,
+        client_relative_path: Vc<FileSystemPath>,
+    ) -> Result<Vc<Box<dyn OutputAsset>>> {
+        let client_relative_path_ref = &*client_relative_path.await?;
+
+        #[derive(Serialize, Default, Debug)]
+        #[serde(rename_all = "camelCase")]
+        pub struct SerializedBuildManifest {
```

Comment:
You could use lifetimes and `&str` here to avoid cloning the RcStrs:

`SerializedBuildManifest<'l>` and `root_main_files: Vec<&'l str>`

but it's probably not worth it to refactor it...